core/src/main/java/com/scalar/db/exception/transaction/TransactionException.java

```diff
@@ -1,13 +1,29 @@
 package com.scalar.db.exception.transaction;
 
+import java.util.Optional;
+import javax.annotation.Nullable;
+
 /** Base class for all exceptions thrown by the Transaction API. */
 public class TransactionException extends Exception {
 
-  public TransactionException(String message) {
+  @Nullable private final String transactionId;
```

Comment:
Added the `transactionId` field to `TransactionException` that's a base class for all exceptions thrown by the Transaction API.